app/src/main/java/vn/com/tma/idlesmart/MainActivity.java

```diff
@@ -2799,21 +2807,36 @@ public void openAlertDialog(int faultId) {
     public void closeAlertDialog(int faultId) {
         if (faultId == 0) {
             closeAllAlertDialogs();
-        } else if (this.alertDialog != null) {
-            if (this.alertDialog.isShowing()) {
-                this.alertDialog.dismiss();
+        } else {
+            String faultMessage = this.faults.aFaultMessage[faultId];
+            String faultDesc = this.faults.aFaultDesc[faultId];
+
+            Bundle bundle = new Bundle();
+            bundle.putInt("faultId", faultId);
+            bundle.putString("faultMessage", faultMessage);
+            bundle.putString("faultDesc", faultDesc);
+
+            AlertDialogFragment alertDialogFragment = new AlertDialogFragment();
+            alertDialogFragment.setArguments(bundle);
+
+            if (alertDialogFragment.getShowsDialog()) {
+                alertDialogFragment.dismiss();
             }
             back2sleep();
         }
     }
 
     public void closeAllAlertDialogs() {
-        if (this.alertDialog != null) {
-            if (this.alertDialog.isShowing()) {
-                this.alertDialog.dismiss();
-            }
-            back2sleep();
+
+        Bundle bundle = new Bundle();
+        bundle.putInt("faultId", 0);
+        bundle.putString("faultMessage", "");
+        bundle.putString("faultDesc", "");
```

Comment:
We no need to send a `bundle` with these arguments if we will not use them. We should check arguments exist in `AlertDialogFragment`